wazo_webhookd/services/mobile/plugin.py

```diff
@@ -0,0 +1,213 @@
+            )
+            for subscription in subscriptions:
+                self.subscription_service.delete(subscription.uuid)
+            logger.info('User unregistered: %s/%s', tenant_uuid, user_uuid)
+
+    def get_tenant_uuid(self, user_uuid):
+        auth = self.get_auth(self._config)
+        return auth.users.get(user_uuid)["tenant_uuid"]
+
+    @classmethod
+    def get_auth(cls, config):
+        auth_config = dict(config['auth'])
+        # FIXME(sileht): Keep the certificate
+        auth_config['verify_certificate'] = False
+        auth = AuthClient(**auth_config)
+        token = auth.token.new('wazo_user', expiration=3600)
+        auth.set_token(token["token"])
+        auth.username = None
+        auth.password = None
+        return auth
+
+    @classmethod
+    def get_external_token(cls, config, user_uuid):
+        auth = cls.get_auth(config)
+        token = auth.external.get('mobile', user_uuid)
+        tenant_uuid = auth.users.get(user_uuid)['tenant_uuid']
+        external_config = auth.external.get_config('mobile', tenant_uuid)
+
+        if token["apns_token"]:
+            external_config['ios_apns_cert'] = '/tmp/ios.pem'
+
+            with open(external_config['ios_apns_cert'], 'w') as cert:
+                cert.write(external_config['ios_apn_certificate'] + "\r\n")
+                cert.write(external_config['ios_apn_private'])
+
+        return (token, external_config)
+
+    @classmethod
+    def run(cls, task, config, subscription, event):
+        user_uuid = subscription['events_user_uuid']
+        # TODO(sileht): We should also filter on tenant_uuid
+        # tenant_uuid = subscription.get('events_tenant_uuid')
+        if (event['data'].get('user_uuid') == user_uuid
+                # and event['data']['tenant_uuid'] == tenant_uuid
+                and event['name'] in ['chatd_user_room_message_created',
+                                      'call_created']):
+            return
+
+        data, external_config = cls.get_external_token(config, user_uuid)
+        token = data['token']
+        apns_token = data['apns_token']
+        push = PushNotification(token, apns_token, external_config)
+
+        msg = None
+        data = event.get('data')
+        name = event.get('name')
+
+        if name == 'user_voicemail_message_created':
+            msg = dict(notification_type='voicemailReceived', items=data)
+
+        if name == 'call_push_notification':
+            msg = dict(notification_type='incomingCall', items=data)
+
+        if name == 'chatd_user_room_message_created':
+            msg = dict(notification_type='messageReceived', items=data)
+
+        if msg:
+            return push.send_notification(msg)
+
+
+class PushNotification(object):
+
+    def __init__(self, external_token, apns_token, external_config):
+        self.token = external_token
+        self.apns_token = apns_token
+        self.external_config = external_config
+
+    def send_notification(self, data):
```

Comment:
I'd rather have 3 methods
`send_incoming_call_notification`, `send_voicemail_received_notification`, `send_chat_msg_notification` and avoid the if here